### ISSUE TITLE
Subscription/Notification with time interval does not work

### ISSUE THREAD

**Issue by Blobonat** [ScorpioBroker/ScorpioBroker#338]
When using the `timeInterval` option in a subscription in order to get periodic notifications Scorpio throws an internal error

I use the same test setup as in #337 (scorpiobroker/all-in-one-runner:spring-kafka-latest )

Steps to reproduce:

1. I create the following entity
```
curl --location --request POST 'http://localhost:9090/ngsi-ld/v1/entityOperations/create' \
--header 'Content-Type: application/json' \
--header 'NGSILD-Tenant: openiot' \
--header 'Link: <https://fiware.github.io/data-models/full-context.jsonld>; rel="http://www.w3.org/ns/json-ld#context"; type="application/ld+json"' \
--data-raw '[
    {
        "id": "urn:ngsi-ld:Vehicle:A500",
        "type": "Vehicle",
        "brandName": {
            "type": "Property",
            "value": "Mercedes"
        },
        "speed": {
            "type": "Property",
            "value": 80
        }
    }
]'
```
2. In order to get a notification every 5 seconds I create the following subscription:
```
curl --location --request POST 'http://localhost:9090/ngsi-ld/v1/subscriptions/' \
--header 'Content-Type: application/json' \
--header 'NGSILD-Tenant: openiot' \
--header 'Link: <https://fiware.github.io/data-models/full-context.jsonld>; rel="http://www.w3.org/ns/json-ld#context"; type="application/ld+json"' \
--data-raw '{
  "id": "urn:ngsi-ld:Subscription:withTimeInterval",
  "type": "Subscription",
  "entities": [{"type": "Vehicle"}],
  "timeInterval": 5,
  "throttling": 0,
  "notification": {
    "format": "normalized",
    "attributes": ["speed", "brandName"],
    "endpoint": {
      "uri": "https://webhook.site/b8930c94-f365-4670-8c6c-467041f27356",
      "accept": "application/ld+json"
    }
  }
}'
```
- Expected behavior: A notification every 5 seconds
- Actual behavior: No notification and the following error log inside Scorpio:
```
2022-10-09 20:31:20.173 ERROR 8 --- [        Timer-9] e.n.n.c.s.NotificationHandlerREST        : Exception ::

java.lang.NullPointerException: Cannot invoke "java.lang.Iterable.iterator()" because "elements" is null
	at com.google.common.collect.Sets.newHashSet(Sets.java:257) ~[guava-29.0-jre.jar!/:na]
	at eu.neclab.ngsildbroker.commons.tools.HttpUtils.generateNotification(HttpUtils.java:674) ~[Commons-1.1.0-SNAPSHOT.jar!/:na]
	at eu.neclab.ngsildbroker.commons.datatypes.Notification.toCompactedJson(Notification.java:101) ~[Commons-1.1.0-SNAPSHOT.jar!/:na]
	at eu.neclab.ngsildbroker.commons.subscriptionbase.NotificationHandlerREST.sendReply(NotificationHandlerREST.java:64) ~[Commons-1.1.0-SNAPSHOT.jar!/:na]
	at eu.neclab.ngsildbroker.commons.subscriptionbase.BaseNotificationHandler.notify(BaseNotificationHandler.java:80) ~[Commons-1.1.0-SNAPSHOT.jar!/:na]
	at eu.neclab.ngsildbroker.commons.subscriptionbase.IntervalNotificationHandler$MyTimer.run(IntervalNotificationHandler.java:74) ~[Commons-1.1.0-SNAPSHOT.jar!/:na]
	at java.base/java.util.TimerThread.mainLoop(Timer.java:566) ~[na:na]
	at java.base/java.util.TimerThread.run(Timer.java:516) ~[na:na]
```

**Comment by ScorpioBroker** [comment]
Hi, 

is fixed in 3.0.5 now. spring is already up on dockerhub, the other version will be up sometime today.

BR

**Comment by Blobonat** [comment]
Thanks a lot!
